### ISSUE TITLE
Consumed Water Heating Today does not report values in Energy Dashboard

### ISSUE THREAD

**Issue by tmb80c** [pail23/stiebel_eltron_isg_component#332]
Stiebel Eltron ISG Consumed Heating Today reports the daily consumed kwh of electricity used to produce the heating of the day in the energy dashboard.

Stiebel Eltron ISG Consumed Water Heating Today should do the same for the water heating but it doesn't report the values.

I tested the entities under "Einzelne Geräte" - Single Devices.

**Comment by janwing** [comment]
The ISG web reports only complete kWh on Consumed Water Heating Today, so if you use only 0.9 per day, it does not report anything on that day.  I use "Stiebel Eltron ISG Consumed Water Heating Total" in the Single Devices Section. ISG web only reports the full kWh, but with the total sensor you'll get a reading when the next full kWh is reached.

**Comment by bitte-ein-bit** [comment]
#214 had a feature request hidden in there to increase the precision. On the ISG I see more digits, but I have no understanding of Modbus unfortunately. Is increasing the precision possible?

**Comment by Mysonemo** [comment]
Thanks for that solution it help me as well !!

**Comment by Mysonemo** [comment]
![image](https://github.com/user-attachments/assets/c4ccdd16-a239-49f0-8e99-c5decfcac79d)
I've an issue the value are all 1h late as the first sensor value at 1h05 am.
How did you do ?

**Comment by Padanian** [comment]
Check DST settings in WPM and ISG. You are probably still in summer time or ISG is on a different time shift than WPM

**Comment by Mysonemo** [comment]
> Check DST settings in WPM and ISG. You are probably still in summer time or ISG is on a different time shift than WPM

Already checked on WPM and ISG and all good. On home assistant as well.
So I think it's because the first update happen   after 12am but just a supposition. 
@sebcou21 How does it fit for you ?

**Comment by Mysonemo** [comment]
You said to modify the time of the ISG but in fact it's the time of the heat pump that I need to change.
________________________________

**Comment by Mysonemo** [comment]
So I've put 5min more (so the heat pump is in advance) and im polling every 90 minutes.
Looks good 
![Screenshot_20241214_081009_Home Assistant](https://github.com/user-attachments/assets/7a7b63f1-04b9-4c1b-aee0-0ab385749850)

Thank's

**Comment by Mysonemo** [comment]
> > You're welcome. But I'm not happy to have to set the scan_interval to 60 seconds for this to work. [This post](https://community.home-assistant.io/t/set-time-to-scrape/102193/3) provides a good way of updating the sensor manually, I will give it a go when I have some time.
> 
> So I finally managed to do that. My ISG is configured to be 3 minutes ahead. I'm using Node-RED and the cronplus node to update the sensor every hour at :59 except at 23:59 or the counter gets reset to 0 before midnight and it messes the energy graph up a lot. Instead, I do it at 00:01.
> 
> Another problem I discovered is that consumption between 23:00 and 23:59 will always be lost because the value on the ISG is updated after the hour and gets reset at midnight. The consumption will appear as untracked in the energy graph. I challenged a few ideas but I currently see no way around that.
> 
> Please let me know if you see the same behavior. Thanks.

Yes that's what happend on my side as well. The consumption is always missing between 23 and 00  as it's cleared before the update so look's a bug to me. I can try to report it because it's really not reporting the comsumption of the day.
